Update parent pom bom gson versions

raahikwueto opened the pull request:
<!-- Please describe your pull request here. -->

update  base Jenkins and BOM to 2.401.x,  GSON to 2.10.1, lang3 to 3.12.0 and changes to imported test library 

```
### Submitter checklist
- [x ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ x] Ensure that the pull request title represents the desired changelog entry
- [x ] Please describe what you did

```
![build result](https://github.com/jenkinsci/ironmq-notifier-plugin/assets/57661980/2ca11fc4-70bc-4245-a5da-d1a926ac8de2)
![clean verify](https://github.com/jenkinsci/ironmq-notifier-plugin/assets/57661980/7a253e12-885f-48a4-8946-8cfaca6ea344)
![plugin running on jenkins ](https://github.com/jenkinsci/ironmq-notifier-plugin/assets/57661980/0578c374-db4b-4e73-a954-e8ea53f06410)
![Test results](https://github.com/jenkinsci/ironmq-notifier-plugin/assets/57661980/b9e6e760-7028-453b-a300-3a632aeb2259)

raahikwueto committed:
update parent pom to 4.76

raahikwueto committed:
update parent pom, base jenkins, bom and gson versions

MarkEWaite commented on pom.xml:
```diff
@@ -69,12 +69,12 @@ limitations under the License.
 
     <artifactId>ironmq-notifier</artifactId>
-    <version>1.0.20-SNAPSHOT</version>
+    <version>1.0.19</version>
```
You don't want to change this.  Keep it as a snapshot version until you are ready to switch to continuous delivery.
```suggestion
    <version>1.0.20-SNAPSHOT</version>
```

MarkEWaite commented on pom.xml:
```diff
@@ -159,19 +158,18 @@ limitations under the License.
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-lang3</artifactId>
-            <version>3.9</version>
+            <version>3.12.0</version>
```
This can be replaced with the [commons lang3 api plugin](https://plugins.jenkins.io/commons-lang3-api/dependencies/)

MarkEWaite commented on pom.xml:
```diff
@@ -159,19 +158,18 @@ limitations under the License.
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
-            <version>2.8.2</version>
+            <version>2.10.1</version>
```
This can be replaced with the [gson api plugin](https://plugins.jenkins.io/gson-api/dependencies)

MarkEWaite reviewed (commented):
Thanks for improving the plugin!

raahikwueto committed:
Implementing review suggestions

raahikwueto commented:
I've Implemented the suggested changes

MarkEWaite reviewed (commented):
Reformatting the pom file is usually best done as a separate step that only performs the reformat.  Reformatting the pom file is also usually best left as a decision for the maintainer of the plugin rather than for someone who is not the maintainer.

There are automated ways to format the pom file and the source code so that maintainers don't need to worry about formatting.   The parent pom of this plugin includes the support for the command `mvn spotless:apply`.  However, that is usually best done by a maintainer when they decide that they want to switch to automated formatting.

This plugin is [up for adoption](https://www.jenkins.io/doc/developer/plugin-governance/adopt-a-plugin/).  @raahikwueto if you are interested in adopting the plugin, it would be a great help for the plugin.  That would make you a maintainer, then you could apply this reformatting and could even apply reformatting to the rest of the source files in the repository.  Maintaining a plugin is not a lot of effort and there are several people like me that are willing to coach new maintainers.

I've submitted a proposed change to replace the transitive dependencies on outdated versions of commons-lang3 and gson jar files with an explicit dependency on the Jenkins commons-lang3 plugin and the Jenkins gson plugin.  Refer to my proposed changes to this pull request in:

* https://github.com/raahikwueto/ironmq-notifier-plugin/pull/2

MarkEWaite committed:
Complete the reformatting

MarkEWaite committed:
Depend on gson and commons-lang3 API plugins

The transitive dependencies included through the ironmq jar file are out
of date and won't be used in Jenkins installations where the API plugins
are already installed.  That will be the case in most Jenkins
controllers, so let's make this plugin use the API plugins rather than
packaging the commons-lang3 and gson jar files as transitive
dependencies.

The commons-lang3 plugin is installed on 65% of all Jenkins controllers
and is delivering the most recent release of commons-lang3.

The gson plugin is a new enough plugin that we don't have installation
count data yet, but it is expected to be widely installed as well,
since it is used in the Trilead API plugin that is installed on 93%
of Jenkins controllers.

raahikwueto committed:
Merge pull request #2 from MarkEWaite/update-parent-pom-bom-gson-versions

Use gson and commons-lang3 API plugins instead of jar files

MarkEWaite reviewed (approved):
Changes look good to me.  Thanks very much!

raahikwueto commented:
Thank you @MarkEWaite  for the  guidance. I'll follow the Jenkins guideline for adopting a plugin
